src/preferences/general.hpp

```diff
@@ -326,4 +326,7 @@ namespace preferences {
 	 */
 	void set_sub_achievement(const std::string& content_for, const std::string& id, const std::string& sub_id);
 
+	void set_editor_chosen_addon(const std::string& addon_id);
```

Comment:
Documentation would be good, because this is a general file where people will be reading it while thinking about other contexts than the new addon-aware editor.

Reply:
Added.

Reply:
Looks good, and I feel the docs for `editor_chosen_addon` cover what I wanted for both functions. I'm happy for the PR to merge as-is.

Suggestion for the setter: The add-on chosen when the editor was last opened (or changed to later within the editor). Will be left set after exiting the editor, so that it can be used as a default next time.

Reply:
I'm not sure what you mean by
> or changed to later within the editor

Reply:
I'm assuming the editor could have a way to select a different add-on without exiting and reopening the editor.

Reply:
It intentionally does not.